tbtaf/main.py

```diff
@@ -0,0 +1,179 @@
+from __future__ import absolute_import
+import os
+import sys
+sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
+from fastapi import FastAPI, HTTPException, BackgroundTasks
+from uuid import uuid4
+from typing import Dict, Optional, List
+from pydantic import BaseModel
+import time
+from orchestrator.TBTAFOrchestrator import TBTAFOrchestrator
+from interpreter.TBTAFInterpreter import TBTAFInterpreter
+from databridge.TBTAFSqliteDatabridge import TBTAFSqliteDatabridge
+
+app = FastAPI()
+
+# Mockup for demonstration - replace with actual TBTAF integration
+class JobStatus(BaseModel):
+    status: str  # "queued", "in_progress", "completed", "failed"
+    progress: int
+    results: Optional[List[Dict]] = None
+    suite_name: str
+
+
+# In-memory storage for demonstration purposes
+jobs_db: Dict[str, JobStatus] = {}
+suites_db: Dict[str, object] = {}  # Global storage for suites by job_id
+
+
+targetDatabridge = TBTAFSqliteDatabridge()
+orch = TBTAFOrchestrator(targetDatabridge=targetDatabridge)
+interpreter = TBTAFInterpreter()
+
+# Pydantic models
+class OperationRequest(BaseModel):
+    suite_name: str
+    tags: Optional[List[str]] = None
+    params: Optional[Dict] = None
+
+class JobStatusResponse(BaseModel):
+    job_id: str
+    status: str
+    progress: int
+    results: Optional[List[Dict]] = None
+
+
+class JobResultResponse(BaseModel):
+    job_id: str
+    results: Optional[List[Dict]] = None
+    summary: str
+
+class SuitesResponse(BaseModel):
+    project_name: str
+    suites: Optional[Dict] = None
+
+def execute_suite_async(job_id: str, suite_name: str):
+    time.sleep(5)
+
+    try:
+        # Get test suite from TBTAF
+        test_suite = orch.createTestSuite("./test/discoverer/samples", testSuiteID=job_id)
```

Comment:
Let´s add some argument within the request (contained in the params) in order to optionally specify the test suite path. If the argument is not present, then yes, we default to test/discoverer/samples